tests/ops/fixtures/saas/rollbar_fixtures.py

```diff
@@ -0,0 +1,272 @@
+        )
+        return item_response
+
+
+@pytest.fixture(scope="function")
+def rollbar_test_client(
+    rollbar_connection_config: RollbarTestClient,
+) -> Generator:
+    test_client = RollbarTestClient(rollbar_connection_config=rollbar_connection_config)
+    yield test_client
+
+
+def _project_exists(project_id: str, rollbar_test_client: RollbarTestClient) -> Any:
+    """ """
+    project_response = rollbar_test_client.get_project(project_id=project_id)
+
+    if not project_response.status_code == 404 and project_response.json()["result"]:
+        return project_response.json()
+
+
+def _item_exists(project_tokens, rollbar_test_client: RollbarTestClient) -> Any:
+    """ """
+    item_response = rollbar_test_client.get_item(project_tokens=project_tokens)
+
+    if not item_response.status_code == 404 and item_response.json()["result"]:
+        return item_response.json()
+
+
+@pytest.fixture(scope="function")
+def rollbar_erasure_data(
+    rollbar_test_client: RollbarTestClient, rollbar_erasure_identity_email: str
+) -> Generator:
+    """
+    Creates a dynamic test data record for erasure tests.
+    """
+    """
+        there are 3 steps to create data for erasure api
+        1) create a new project
+        2) get access token with scope post_server_item from project details
+        3) create item inside above created project
+    """
+
+    # 1) create a new project
+    project_response = rollbar_test_client.create_project()
+    project = project_response.json()
+    project_id = project["result"]["id"]
+
+    error_message = (
+        f"Project with project id [{project_id}] could not be added to Rollbar"
+    )
+    poll_for_existence(
+        _project_exists,
+        (project_id, rollbar_test_client),
+        error_message=error_message,
+    )
+
+    # 2) get access token with scope post_server_item, read from project details
+    project_access_token_response = rollbar_test_client.get_project(
+        project_id=project_id
+    )
+    access_tokens = project_access_token_response.json()
+    access_tokens_result = access_tokens["result"]
+    # fetch only specific tokens from token list
+    project_tokens = {"read": "", "post_server_item": ""}
+    for token in access_tokens_result:
+        if token["name"] == "read":
+            project_tokens["read"] = token["access_token"]
+        if token["name"] == "post_server_item":
+            project_tokens["post_server_item"] = token["access_token"]
+
+    # 3) create item inside above created project
+    item_response = rollbar_test_client.create_item(
+        project_tokens, rollbar_erasure_identity_email
+    )
+    item = item_response.json()["result"]
+    error_message = (
+        f"Item within project id [{project_id}] could not be added to Rollbar"
+    )
+    poll_for_existence(
+        _item_exists,
+        (project_tokens, rollbar_test_client),
+        error_message=error_message,
+    )
+    yield item
```

Comment:
Let's delete the project we generated here after the tests are complete. In general we want to clean up all the data we create as part of tests.